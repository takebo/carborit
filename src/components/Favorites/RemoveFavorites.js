import React from 'react';
import PropTypes from 'prop-types';

const RemoveFavorites = props => {
  return (
    <button
      className="remove-favorites"
      value={props.value}
      onClick={event => props.deleteFav(event.target.value)}
    >
      remove
    </button>
  );
};

export default RemoveFavorites;

RemoveFavorites.propTypes = {
  value: PropTypes.string.isRequired,
  deleteFav: PropTypes.func.isRequired,
};
