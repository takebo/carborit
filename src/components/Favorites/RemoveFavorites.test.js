import React from 'react';
import { shallow } from 'enzyme';

import RemoveFavorites from './RemoveFavorites';

describe('RemoveFavorites', () => {
  it('should render', () => {
    const props = { value: 'model+year', deleteFav: jest.fn() };
    const component = shallow(<RemoveFavorites {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('button click should delete favorite', () => {
    const props = { value: 'model+year', deleteFav: jest.fn() };
    const component = shallow(<RemoveFavorites {...props} />);
    component.simulate('click', { target: { value: 123 } });
    expect(props.deleteFav).toHaveBeenCalled();
  });
});
