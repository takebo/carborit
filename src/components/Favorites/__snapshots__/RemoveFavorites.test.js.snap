// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveFavorites should render 1`] = `
<button
  className="remove-favorites"
  onClick={[Function]}
  value="model+year"
>
  remove
</button>
`;
